lib/phoenix/controller.ex

```diff
@@ -206,16 +217,29 @@ defmodule Phoenix.Controller do
 
   @doc """
   Stores the view for rendering.
+
+  Raises `Phoenix.ControllerAlreadySentError` if the conn was already sent.
   """
   @spec put_view(Plug.Conn.t, atom) :: Plug.Conn.t
+  def put_view(%Plug.Conn{state: state}, _module) when not state in @unsent do
```

Comment:
Since the final states in Plug are either `:sent` or `:chunked`, should change it to:

```elixir
@sent [:chunked, :sent]
```

And change the guards as @lancehalvorsen suggested above.